docs/usage/debugging.rst

```diff
@@ -0,0 +1,116 @@
+Debugging
+=========
+
+Using the Python debugger
+--------------------------
+
+You can configure Litestar to drop into the :doc:`Python Debugger <python:library/pdb>`
+when an exception occurs. This can be configured in different ways:
+
+Configuring ``Litestar`` with the ``pdb_on_exception`` option
+    .. code-block:: python
+
+        app = Litestar(pdb_on_exception=True)
+
+
+Running your app with the CLI and using the ``--pdb`` flag
+    .. code-block:: shell
+
+        litestar run --pdb
+
+Using the ``LITESTAR_PDB`` environment variable
+    ``LITESTAR_PDB=1``
+
+
+Debugging with an IDE
+---------------------
+
+You can easily attach your IDEs debugger to your application, whether you're running it
+via the CLI or a webserver like `uvicorn <https://www.uvicorn.org/>`_.
+
+Intellij / PyCharm
+++++++++++++++++++
+
+Using the CLI
+*************
+
+1. Create a new debug configuration via ``Run`` > ``Edit Configurations``
+2. Select ``Module name`` option and set it to ``litestar``
+3. Add the ``run`` parameter and optionally additional parameters you want to pass
+   to the CLI
+
+   .. image:: /images/debugging/pycharm-config-cli.png
+
+4. Run your application in the debugger via ``Run`` > ``Debug Litestar``
+
+   .. image:: /images/debugging/pycharm-debug.png
+        :align: center
+
+
+.. important::
+    Breakpoints inside route handlers might not work correctly when used in conjunction
+    with the ``--reload`` and ``--web-concurrency`` parameters. If you want to use the
+    CLI while making use of these options, you can attach the debugger manually to the
+    running uvicorn process via ``Run`` > ``Attach to process``.
+
+
+Using uvicorn
+*************
+
+1. Create a new debug configuration via ``Run`` > ``Edit Configurations``
+2. Select ``Module name`` option and set it to ``uvicorn``
+3. Add the ``app:app`` parameter (or the equivalent path to your application object)
+
+   .. image:: /images/debugging/pycharm-config-uvicorn.png
+
+4. Run your application in the debugger via ``Run`` > ``Debug Litestar``
+
+   .. image:: /images/debugging/pycharm-debug.png
+        :align: center
+
+
+VS Code
++++++++
+
+
+Using the CLI
+*************
+
+
+1. Open the debugging configurations via ``Run`` > ``Open configurations``
+   .. image:: /images/debugging/vs-code-config.png
+2. Add the following configuration:
+    .. code-block:: json
+
+        {
+            "name": "Python: Litestar app",
+            "type": "python",
+            "request": "launch",
+            "module": "litestar",
+            "justMyCode": true,
+            "args": ["run"]
+        }
+
+3. Run your application via the debugger via ``Run`` > ``Start debugging``
+    .. image:: /images/debugging/vs-code-debug.png
```

Comment:
I don't know if this is just b/c I'm such a vscode newb, but when I run this, the debugger isn't selecting the poetry interpreter for the project. Rather, its selecting an alpha version of 3.12 which may be the latest pyenv version I have available, which results in `No module named litestar`.

Any @litestar-org/maintainers or @litestar-org/members with better vscode foo out there who wants to confirm this config and/or tell me what I might be doing wrong?

Reply:
No idea to be honest. For me it just picked up the virtualenv. I tried it with poetry and without, worked just fine. Maybe it's some extension thing?

Anyway, I suspect this is not specific to our configuration, but rather about how you set up VS Code to handle environments, so I guess it's best to leave that out of this guide since there are way to many options. What do you think?

Reply:
Yes, very likely to be a problem at my end!